web/Dockerfile

```diff
@@ -7,10 +7,12 @@ RUN mkdir -p $APP_DIR/src
 WORKDIR $APP_DIR
 
 RUN npm install -g @angular/cli@6.1.3 --silent --depth 1
-COPY package.json $APP_DIR
-RUN npm install --silent --depth 0
 
-COPY angular.json karma.conf.js protractor.conf.js tsconfig.json tslint.json $APP_DIR
+COPY package.json angular.json karma.conf.js protractor.conf.js tsconfig.json tslint.json $APP_DIR
 COPY ./src $APP_DIR
 
+VOLUME /usr/src/npm-cache
+RUN npm install --depth 0
```

Comment:
This is problematic. The dependencies should be installed _before_ the source is copied. Otherwise, the image has to be entirely rebuilt for every code change.